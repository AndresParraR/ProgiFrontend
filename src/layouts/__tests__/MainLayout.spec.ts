import { mount } from "@vue/test-utils";
import { beforeEach, describe, expect, test } from "vitest";
import { Quasar } from "quasar";
import MainLayout from "../MainLayout.vue";

describe("MainLayout Component", () => {
  let wrapper: any = null;

  beforeEach(() => {
    const global = {
      plugins: [Quasar],
    };

    wrapper = mount(MainLayout, {
      global,
    });
  });

  test("Should render correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
