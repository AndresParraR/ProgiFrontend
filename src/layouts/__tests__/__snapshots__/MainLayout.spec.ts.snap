// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainLayout Component > Should render correctly 1`] = `
<div
  class="q-layout q-layout--standard"
  style="min-height: 768px;"
  tabindex="-1"
>
  <!---->
  <object
    aria-hidden="true"
    class="q--avoid-card-border"
    data="about:blank"
    style="display: block; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
    tabindex="-1"
    type="text/html"
  />
  <header
    class="q-header q-layout__section--marginal fixed-top"
  >
    <div
      class="q-toolbar row no-wrap items-center"
      role="toolbar"
    >
      <div
        class="q-toolbar__title ellipsis"
      >
         Progi App 
      </div>
    </div>
    <div
      class="q-layout__shadow absolute-full overflow-hidden no-pointer-events"
    />
    <object
      aria-hidden="true"
      class="q--avoid-card-border"
      data="about:blank"
      style="display: block; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; height: 100%; width: 100%; overflow: hidden; pointer-events: none; z-index: -1;"
      tabindex="-1"
      type="text/html"
    />
  </header>
  <div
    class="q-page-container"
    style="padding-top: 50px;"
  >
    <router-view />
  </div>
</div>
`;
